src/main/java/LoginServlet.java

```diff
@@ -0,0 +1,35 @@
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
+      jsonResponse.addProperty("loggedIn", false);
+      // After login redirect user to nickname-input page
+      String loginUrl = userService.createLoginURL("/");
+      jsonResponse.addProperty("loginUrl", loginUrl);
+    } else {
+      jsonResponse.addProperty("loggedIn", true);
+      String logoutUrl = userService.createLogoutURL("/");
+      jsonResponse.addProperty("logoutUrl", logoutUrl);
```

Comment:
I can't seem to log out. When I try, the page says "Redirecting you to https://accounts.google.com/Logout?continue=https://uc.appengine.google.com/_ah/logout%3Fcontinue%3Dhtt..." (there's much more to this URL), then it opens my Chrombook's settings. But the settings don't have a way to log out as far as I can tell.

Also, it would be nice if it were clear "how" people were logging in when they click log in - when I clicked, the page just refreshed and said "log out". There was no indication that it was my google account that had been logged in.

Reply:
^^